privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -399,7 +399,7 @@ def getUserList(self, searchDict):
                 comperator = ">="
                 if searchDict[search_key] in ["1", 1]:
                     comperator = "<="
-                filter += "(|({0!s}{1!s}{2!s})({3!s}!=0))".format(self.userinfo[search_key],
+                filter += "(&({0!s}{1!s}{2!s})(!({3!s}=0)))".format(self.userinfo[search_key],
                                                   comperator,
```

Comment:
This is due to the only valid condition matchers for a ldap search filter are "~= = >= <=", not conditions are expressed as (!({3!s}=0)).

Best Regards

Martin